Client/mods/deathmatch/logic/CClientPlayerClothes.cpp

```diff
@@ -464,121 +349,163 @@ const char* CClientPlayerClothes::GetClothingName(unsigned char ucType)
-            case 5:
-                return LEFT_LOWER_ARM_CLOTHING_MAX;
-            case 6:
-                return RIGHT_UPPER_ARM_CLOTHING_MAX;
-            case 7:
-                return RIGHT_LOWER_ARM_CLOTHING_MAX;
-            case 8:
-                return BACK_TOP_CLOTHING_MAX;
-            case 9:
-                return LEFT_CHEST_CLOTHING_MAX;
-            case 10:
-                return RIGHT_CHEST_CLOTHING_MAX;
-            case 11:
-                return STOMACH_CLOTHING_MAX;
-            case 12:
-                return LOWER_BACK_CLOTHING_MAX;
-            case 13:
-                return EXTRA1_CLOTHING_MAX;
-            case 14:
-                return EXTRA2_CLOTHING_MAX;
-            case 15:
-                return EXTRA3_CLOTHING_MAX;
-            case 16:
-                return EXTRA4_CLOTHING_MAX;
-            case 17:
-                return SUIT_CLOTHING_MAX;
+            return false;
         }
+
+        clothes.push_back({texture, model});
+        return true;
     }
 
-    return 0;
+    return false;
 }
 
-bool CClientPlayerClothes::IsValidModel(unsigned short usModel)
+bool CClientPlayerClothes::RemoveClothingModel(const char* texture, const char* model, unsigned char clothingType)
 {
-    return usModel >= CLOTHES_MODEL_ID_FIRST && usModel <= CLOTHES_MODEL_ID_LAST;
+    if (clothingType < PLAYER_CLOTHING_SLOTS)
+    {
+        if (texture == nullptr || model == nullptr)
+            return false;
+
+        auto& clothes = m_NewClothes[clothingType];
+
+        auto it = std::find_if(clothes.begin(), clothes.end(),[&](const SPlayerClothing& clothing) {
+            return !stricmp(texture, clothing.szTexture.c_str()) && !stricmp(model, clothing.szModel.c_str());
+        });
+
+        if (it == clothes.end())
+            return false;
+
+        clothes.erase(it);
+        m_bHasClothesChanged = true;
+
+        return true;
+    }
+
+    return false;
+}
+
+bool CClientPlayerClothes::HasClothesChanged()
+{
+    return m_bHasClothesChanged;
+}
+
+void CClientPlayerClothes::RefreshClothes()
+{
+    for (std::uint8_t clothingType = 0; clothingType < PLAYER_CLOTHING_SLOTS; clothingType++)
+    {
+       auto& clothes = m_NewClothes[clothingType];
+
+       if (clothes.empty() && !m_bHasClothesChanged)
+           continue;
+
+       bool                   hasInvalidClothing = false;
+       const SPlayerClothing* pCurrent = m_Clothes[clothingType];
+
+       if (!m_bHasClothesChanged)
+       {
+           for (auto clothing = clothes.begin(); clothing != clothes.end();)
+           {
+               std::string fileTXD = clothing->szTexture + ".txd";
+               std::string fileDFF = clothing->szModel + ".dff";
+
+               if (!g_pGame->GetRenderWare()->HasClothesFile(fileTXD.c_str()) || !g_pGame->GetRenderWare()->HasClothesFile(fileDFF.c_str()))
+               {
+                   if (pCurrent && (pCurrent->szTexture == clothing->szTexture || pCurrent->szModel == clothing->szModel))
```

Comment:
Is it correct to compare pointers instead of `strcmp` in this case?

Reply:
Ahh, I see. You need to change the prefix `sz`, because it means a null-terminated C string.